storage/types.go

```diff
@@ -309,6 +309,11 @@ type Shard interface {
 type databaseShard interface {
 	Shard
 
+	// OnEvictedFromWiredList is the same as block.Owner. Had to duplicate
+	// it here because mockgen chokes on embedded interfaces sometimes:
+	// https://github.com/golang/mock/issues/10
```

Comment:
hm, did their suggestion of using reflected mode not work too?

Reply:
Yeah I think I tried that as well, I could spend some more time on it though